apps/meteor/app/channel-settings/server/functions/saveRoomTopic.ts

```diff
@@ -20,6 +20,10 @@ export const saveRoomTopic = async function (
 		});
 	}
 
+	const room = await Rooms.findOneById(rid);
+
+	await Room.beforeTopicChange(room!);
```

Comment:
we make sure rid is valid one layer up, but still need to query it here again since otherwise would have to change signature and lots of other calls. Since this is not a frequently called function, decided to not do a refactor-ish, and do it like this. wdyt?

Reply:
We can keep it, but if you identified a refactor necessity, let's create a task to keep track of it.